pkg/drupal/dbadmincheck.go

```diff
@@ -0,0 +1,139 @@
+package drupal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os/exec"
+	"strings"
+
+	"github.com/salsadigitalauorg/shipshape/pkg/shipshape"
+	"github.com/salsadigitalauorg/shipshape/pkg/utils"
+)
+
+const AdminUser shipshape.CheckType = "drupal-admin-user"
+
+// AdminUserCheck fetches all role configurations from the database and verifies
+// they do not have is_admin set to true.
+type AdminUserCheck struct {
+	shipshape.CheckBase `yaml:",inline"`
+	DrushCommand        `yaml:",inline"`
+	// List of role names allowed to have is_admin set to true.
+	AllowedRoles []string `yaml:"allowed-roles"`
+	userRoles    []string
+	roleConfigs  map[string]bool
+}
+
+type roleConf struct {
+	IsAdmin bool `json:"is_admin"`
+	Name string `json:"id"`
+}
+
+// Init implementation for the drush-based user role config check.
+func (c *AdminUserCheck) Init(ct shipshape.CheckType) {
+	c.CheckBase.Init(ct)
+	c.RequiresDb = true
+}
+
+// Merge implementation for AdminUserCheck check.
+func (c *AdminUserCheck) Merge(mergeCheck shipshape.Check) error {
+	adminUserMergeCheck := mergeCheck.(*AdminUserCheck)
+	if err := c.CheckBase.Merge(&adminUserMergeCheck.CheckBase); err != nil {
+		return err
+	}
+
+	c.DrushCommand.Merge(adminUserMergeCheck.DrushCommand)
+	utils.MergeStringSlice(&c.AllowedRoles, adminUserMergeCheck.AllowedRoles)
+	return nil
+}
+
+// getActiveRoles runs the drush command to populate data for the roles config check.
+func (c *AdminUserCheck) getActiveRoles() map[string]string {
+  var err error
+
+	activeRoles := map[string][]byte{}
+	rolesListMap := map[string]string{}
+
+  cmd := []string{"role:list", "--fields=.", "--format=json"}
+  activeRoles["user-roles"], err = Drush(c.DrushPath, c.Alias, cmd).Exec()
+  var pathErr *fs.PathError
+  if err != nil && errors.As(err, &pathErr) {
+    c.AddFail(pathErr.Path + ": " + pathErr.Err.Error())
+  } else if err != nil {
+  		msg := string(err.(*exec.ExitError).Stderr)
+  		c.AddFail(strings.ReplaceAll(strings.TrimSpace(msg), "  \n  ", ""))
+  } else {
+    // Unmarshal roles JSON.
+    err = json.Unmarshal(activeRoles["user-roles"], &rolesListMap)
```

Comment:
and this would change as well
```suggestion
    err = json.Unmarshal(activeRoles, &rolesListMap)
```